server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -577,32 +570,62 @@ public int hashCode() {
     return Objects.hash(file);
   }
 
-  public void setClosed(boolean closed) {
-    this.closed = closed;
-  }
-
   public boolean isDeleted() {
-    return deleted;
-  }
-
-  public void setDeleted(boolean deleted) {
-    this.deleted = deleted;
+    return this.status == TsFileResourceStatus.DELETED;
   }
 
   public boolean isCompacting() {
-    return isCompacting;
-  }
-
-  public void setCompacting(boolean compacting) {
-    isCompacting = compacting;
+    return this.status == TsFileResourceStatus.COMPACTING;
   }
 
   public boolean isCompactionCandidate() {
-    return compactionCandidate;
+    return this.status == TsFileResourceStatus.COMPACTION_CANDIDATE;
   }
 
-  public void setCompactionCandidate(boolean compactionCandidate) {
-    this.compactionCandidate = compactionCandidate;
+  public boolean setStatus(TsFileResourceStatus status) {
+    boolean success = false;
+    switch (status) {
+      case CLOSED:
+        {
```

Comment:
LGTM and thanks for getting my assigned job done. Just one small style-related thing here. I'm thinking that the curly braces-style switch statement seems a bit unnecessary here and I haven't seen any curly braces-style switch statement in other places of IoTDB yet.

Reply:
resolve